forest-node/ansible/roles/docker/tasks/main.yml

```diff
@@ -0,0 +1,120 @@
+---
+# tasks file for role/docker
+- name: Upgrade all apt packages
+  apt: upgrade=dist force_apt_get=yes
+
+- name: Create a login user
+  user:
+    name: "{{ user }}"
+    state: present
+
+- name: Add public key to authorized_keys
+  authorized_key:
+    user: "{{ user }}"
+    state: present
+    key: "{{ lookup('file', '~/id_rsa.pub') }}"
+
+- name: Deny root from logging in
+  ansible.builtin.lineinfile:
+    dest: /etc/ssh/sshd_config
+    regexp: '^(#)?PermitRootLogin \w*$'
+    line: 'PermitRootLogin no'
+    state: present
+
+- name: Restart SSHD Service 
+  become: true
+  shell: sudo service ssh restart 
+
+- name: Allow specific users to log in
+  ansible.builtin.lineinfile:
+    dest: /etc/ssh/sshd_config
+    regexp: '^AllowUsers'
+    line: 'AllowUsers {{ user }}'
+    state: present
+
+- name: Add {{ user }} to sudoers file
+  ansible.builtin.lineinfile:
+    path: /etc/sudoers
+    regexp: '^{{ user }}'
+    line: '{{ user }} ALL=(ALL) NOPASSWD: ALL'
+    validate: 'visudo -cf %s'
+
+- name: create docker user
+  user:
+    name: '{{ user }}'
+    append: true
+    shell: /bin/bash
+    system: true
+    create_home: false
+
+- name: Install packages to allow apt to use repository over HTTPS
+  package:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - acl
+    - python3-pip 
+    - apt-transport-https
+    - ca-certificates
+    - curl
+    - software-properties-common
+
+- name: Add Docker's official GPG key
```

Comment:
This all can be omitted by using a proper image I linked earlier.

Reply:
sure